auction-house/program/src/receipt/mod.rs

```diff
@@ -0,0 +1,507 @@
+32 + // bookkeeper
+32 + // auction_house
+32 + // buyer
+32 + // metadata
+1 + 32 + // token_account
+1 + 32 + // purchase_receipt
+8 + // price
+8 + // token_size
+1 + // bump
+1 + // trade_state_bump
+8 + // created_at
+1 + 8; // canceled_at
+
+#[account]
+pub struct BidReceipt {
+    pub trade_state: Pubkey,
+    pub bookkeeper: Pubkey,
+    pub auction_house: Pubkey,
+    pub buyer: Pubkey,
+    pub metadata: Pubkey,
+    pub token_account: Option<Pubkey>,
+    pub purchase_receipt: Option<Pubkey>,
+    pub price: u64,
+    pub token_size: u64,
+    pub bump: u8,
+    pub trade_state_bump: u8,
+    pub created_at: i64,
+    pub canceled_at: Option<i64>,
+}
+
+pub const LISTING_RECEIPT_SIZE: usize = 8 + //key
+32 + // trade_state
+32 + // bookkeeper
+32 + // auction_house
+32 + // seller
+32 + // metadata
+1 + 32 + // purchase_receipt
+8 + // price
+8 + // token_size
+1 + // bump
+1 + // trade_state_bump
+8 + // created_at
+1 + 8; // canceled_at;
+
+#[account]
+pub struct ListingReceipt {
+    pub trade_state: Pubkey,
+    pub bookkeeper: Pubkey,
+    pub auction_house: Pubkey,
+    pub seller: Pubkey,
+    pub metadata: Pubkey,
+    pub purchase_receipt: Option<Pubkey>,
+    pub price: u64,
+    pub token_size: u64,
+    pub bump: u8,
+    pub trade_state_bump: u8,
+    pub created_at: i64,
+    pub canceled_at: Option<i64>,
+}
+
+pub const PURCHASE_RECEIPT_SIZE: usize = 8 + //key
+32 + // buyer
+32 + // seller
+32 + // auction_house
+32 + // metadata
+8 + // token_size
+8 + // price
+1 + // bump
+1 + 8; // created_at
+
+#[account]
+pub struct PurchaseReceipt {
+    pub buyer: Pubkey,
+    pub seller: Pubkey,
+    pub auction_house: Pubkey,
+    pub metadata: Pubkey,
+    pub token_size: u64,
+    pub price: u64,
+    pub bump: u8,
+    pub created_at: Option<i64>,
+}
+
+#[derive(Accounts)]
+#[instruction(receipt_bump: u8)]
+pub struct PrintListingReceipt<'info> {
+    #[account(mut)]
+    receipt: UncheckedAccount<'info>,
+    #[account(mut)]
+    bookkeeper: Signer<'info>,
+    system_program: Program<'info, System>,
+    rent: Sysvar<'info, Rent>,
+    instruction: UncheckedAccount<'info>,
+}
+
+pub fn print_listing_receipt<'info>(
+    ctx: Context<'_, '_, '_, 'info, PrintListingReceipt<'info>>,
+    receipt_bump: u8,
+) -> ProgramResult {
+    let receipt_account = &ctx.accounts.receipt;
+    let instruction_account = &ctx.accounts.instruction;
+    let bookkeeper_account = &ctx.accounts.bookkeeper;
+
+    let rent = &ctx.accounts.rent;
+    let system_program = &ctx.accounts.system_program;
+    let clock = Clock::get()?;
+
+    let prev_instruction = get_instruction_relative(-1, instruction_account)?;
+    let prev_instruction_accounts = prev_instruction.accounts;
+
+    let wallet = &prev_instruction_accounts[0];
+    let auction_house = &prev_instruction_accounts[4];
+    let seller_trade_state = &prev_instruction_accounts[6];
+    let metadata = &prev_instruction_accounts[2];
+
+    let mut buffer = &prev_instruction.data[8..];
```

Comment:
The 1st 8 bytes are for the ID of the instruction so drop those bytes out before trying to deserialize the instruction data.